### PR TITLE
Added http proxy configuration

### PR DESCRIPTION
**Pull request by aecio**
Solves issue #8

### CODE REVIEW

**Comment by kkrugler** [comment]
Hi @aecio - until (or if) we get rid of BaseHttpFetcher, I think it would be more consistent to add the setProxy() call to BaseHttpFetcher, and just use the (protected) proxy variable in SimpleHttpFetcher when building the client.

**Comment by aecio** [comment]
@kkrugler Just fixed that.
I agree BaseHttpFetcher could removed, I see no use for it.

**Comment by kkrugler** [comment]
Hi @aecio - the main value I see for BaseHttpFetcher is if someone wanted to create a radically different fetcher (still based on HttpClient, e.g. the async version). Then having this base class to store settings common across both versions would be better.

There's a minor value in that it makes it easier to create a fake fetcher (if you don't want to real mocking support), which I've done in the past when building fetchers that use this code.

**Comment by aecio** [comment]
@kkrugler - I see your point. I usually prefer to use interfaces and composition for these cases. I have more than one "HTTP fetcher" implementation in the crawler I work on, but currently I just use `BaseFetcher`. I use it mainly because of the fetcher signature (it could be just an interface instead of a abstract class), I find the configurations too specific for the  `SimpleHttpFetcher` implementation. But anyway, it may be useful for other cases.

**Comment by kkrugler** [comment]
Sorry, should have noticed this before...minor nit:

Use leading "_" for instance variables (so proxy => _proxy).

Hmm, wonder if I've got what it takes to add checkstyle to the pom?

**Comment by afs** [comment]
An alternative is to enable system properties.  The change is `HttpClientBuilder.create().useSystemProperties()` to pick up the "http.proxy" settings.

**Comment by aecio** [comment]
@kkrugler Just pushed a fix for the leading `_` formatting problem. Sorry for taking so long, I completely forgot about this.

File: src/main/java/crawlercommons/fetcher/http/BaseHttpFetcher.java [hunk 1]
```diff
@@ -109,4 +112,8 @@ public RedirectMode getRedirectMode() {
         return _redirectMode;
     }
 
+    public void setProxy(String scheme, String host, int port) {
+        this._proxy = new HttpHost(host, port, scheme);
```
Comment (kkrugler) [inline src/main/java/crawlercommons/fetcher/http/BaseHttpFetcher.java]: No need for `this.`, which is one of the reasons we use a leading `_`.

Also would be good to have a `getProxy()` call.

File: src/main/java/crawlercommons/fetcher/http/SimpleHttpFetcher.java [hunk 2]
```diff
@@ -909,6 +911,11 @@ private void init() {
                 requestConfigBuilder.setConnectTimeout(_connectionTimeout);
                 requestConfigBuilder.setConnectionRequestTimeout(_connectionRequestTimeout);
 
+                if(_proxy != null){
```
Comment (kkrugler) [inline src/main/java/crawlercommons/fetcher/http/SimpleHttpFetcher.java]: Minor nit - consistent formatting would be `if (_proxy != null) {`

**Review by kkrugler — CHANGES_REQUESTED** [review]
Some minor nits, but we also need an update to CHANGES.txt

**Pull request merged**